crates/gen/src/windows.rs

```diff
@@ -1,11 +1,25 @@
 /// Returns the build's `.windows` directory in the root of the workspace as a `PathBuf`.
 pub fn build_windows_dir() -> std::path::PathBuf {
-    let mut path = workspace_dir();
+    build_windows_dir_at_workspace(&workspace_dir())
+}
+
+/// Returns the build's `.windows` directory in the passed workspace directory as a `PathBuf`.
+pub fn build_windows_dir_at_workspace(workspace_dir: &std::path::PathBuf) -> std::path::PathBuf {
+    let mut path = workspace_dir.clone();
     path.push(".windows");
     path
 }
 
-fn workspace_dir() -> std::path::PathBuf {
+/// Returns the build's target directory in the passed workspace directory as a `PathBuf`.
+pub fn build_target_dir_at_workspace(workspace_dir: &std::path::PathBuf) -> std::path::PathBuf {
+    let mut path = workspace_dir.clone();
+    path.push("target");
+    path
+}
+
+/// Returns the build's workspace directory. Executes 'cargo' command to get it because Cargo
+/// lacks an environment variable for the workspace dir.
+pub fn workspace_dir() -> std::path::PathBuf {
```

Comment:
Note: we should probably cache this call in a static since this is expensive. This would also remove the need to pass the workspace directory into the other functions which would simplify the calling code.

Reply:
ok, placed workspace_dir() result to lazy_static variable